### PR TITLE
Add "Move" option for deleted items in the content tree

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/3769

### Description

This PR simply adds the "Move" action to deleted items in the content tree - see #3769 for more info.

move-deleted-content-items

### CODE REVIEW

**Comment by kjac** [comment]
**Update:** The extra divider above "Delete" has been removed:

image

**Comment by madsrasmussen** [comment]
Hi @kjac, Thanks for the PR. 

I quick thought. Do we need the extra context menu item. Would it make sense to make the restore functionality "fallback" to a move if it fails?

If a restore can't be made -> show a message that we can't automatically restore the node and show a tree where the user can select where the node should go?

**Comment by kjac** [comment]
@madsrasmussen not a bad idea! My only concern is the amount of duplicate code (and/or refactoring) it'll require to implement this - perhaps it'd be better to settle for a "Click here to move it" button instead?

**Comment by kjac** [comment]
@madsrasmussen something along the lines of this:

restore-move-content-wip

**Comment by madsrasmussen** [comment]
To me, it feels like a workaround :) 

I don't think the code duplication is that bad compared to adding the whole move dialog. If I remember correctly the restore functionality is just a move method so we need to add some extra code to handle the tree.

**Comment by kjac** [comment]
@madsrasmussen you drive a hard bargain 😄 

The move dialog wouldn't be duplicated - it'd be reused. But you're right, it *is* a workaround. In the end the UX is much better if the move operation is embedded within the restore dialog. We'll just have to live with the duplicate code.

Here's a working demo:

restore-content-with-move

**Comment by poornimanayar** [comment]
Hey @kjac @madsrasmussen ,

I have been testing this PR. Quick question, given that we have the restore modal helps you manually move an item when an automatic restore cannot be achieved is it necessary to have the move option there at all? Is it not better to have a Restore which handles restore and move? 

Poornima

**Comment by kjac** [comment]
@poornimanayar originally I had removed the "Move" option from the menu. But I'm not sure it makes sense to leave it out. 

Consider the recycle bin on your computer:

- You can choose to restore an item in it to its original location on disk (at least you can on Windows, I can't say for sure on Mac...) - equivalent to the "Restore" menu item. 
- If you don't want it in its original location, you can choose to move the item somewhere else instead, by dragging or using the folder options - equivalent to the "Move" menu item.

In other words, if people don't want the item restored to its original place (maybe they can't remember where it was in the first place?), I see no harm in letting them opt for "Move" instead.

That being said I really don't feel very strongly about it, so I'd be happy to remove "Move" (see what I did there?).

**Comment by poornimanayar** [comment]
Hello @kjac ,

Yes, that makes total sense to me :-) Different ways of seeing the same issue I think! I'll test the actual functionality quickly and comment :-)

Poornima

**Comment by poornimanayar** [comment]
I have tested this functionality and I can confirm it works. One thought I have is should the Restore Notification when using the Restore button say the same message as the restore not available message which appears in the modal? I might be overthinking it here!

Poornima

**Comment by kjac** [comment]
@poornimanayar there we go!

restore-deleted-content-2

**Comment by poornimanayar** [comment]
Thats a good catch , I didnt note it. What I was suggesting it may be make these two consistent , it kinda is. So I am clearly overthinking I think

image
image

**Comment by poornimanayar** [comment]
Yeah lets leave it at that @kjac . I shall test the latest changes you made and let you know!

Poornima

**Comment by nul800sebastiaan** [comment]
This is so smooth, love it! And thanks for all the UX feedback, really great to see the back and forth! 🏅

**Comment by kjac** [comment]
@Shazwazza that's not a problem at all, I'll get on it soon 😄

I kinda suspected that would happen - I for one have been making changes to these dialogs in both branches. The same problem is bound to arise for the media tree equivalent if that's merged in (#3914), so don't spend time trying to merge that manually either. Just let me know and I'll reapply the changes in V8.

And for good measure: That goes for all my V7 changes. If they're giving you a headache in the merge to V8, let me know and I'll rework them in V8. No point in you guys wasting time when I have time to give 👍
